pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateEvaluator.java

```diff
@@ -72,6 +72,27 @@ public interface PredicateEvaluator {
    */
   boolean applySV(int value);
 
+  /**
+   * Apply the predicate to a batch of single-value entries.
+   * Compact matching entries into the prefix of the arrays.
+   *
+   * @param limit How much of the input to consume.
+   * @param docIds The docIds associated with the values - may be modified by invocation.
+   * @param values Batch of dictionary ids or raw values - may be modified by invocation.
+   * @return the index of the first non-matching entry.
+   */
+  default int applySV(int limit, int[] docIds, int[] values) {
+    int matches = 0;
+    for (int i = 0; i < limit; i++) {
+      int value = values[i];
+      if (applySV(value)) {
+        docIds[matches] = docIds[i];
+        values[matches++] = value;
```

Comment:
For optimal performance, we don't need to modify the `values` array

Reply:
Sure, I thought it could be useful in the future, e.g. to aggregate the matching values

Reply:
I've removed this, it can be added back if it's used in the future